lib/install/validate/network.go

```diff
@@ -34,38 +34,46 @@ import (
 	"golang.org/x/net/context"
 )
 
-func (v *Validator) addNetworkHelper(ctx context.Context, conf *config.VirtualContainerHostConfigSpec, netName, epName, contNetName string, def bool) error {
+func (v *Validator) getEndpoint(ctx context.Context, conf *config.VirtualContainerHostConfigSpec,
+	pgName, epName, contNetName string, def bool,
+	ns []net.IP) (*executor.NetworkEndpoint, error) {
 	defer trace.End(trace.Begin(""))
 
-	moid, err := v.networkHelper(ctx, netName)
+	moid, err := v.networkHelper(ctx, pgName)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	if epName != "" {
-		conf.AddNetwork(&executor.NetworkEndpoint{
+	e := &executor.NetworkEndpoint{
+		Common: executor.Common{
+			Name: epName,
+		},
+		Network: executor.ContainerNetwork{
 			Common: executor.Common{
-				Name: epName,
+				Name: contNetName,
+				ID:   moid,
 			},
-			Network: executor.ContainerNetwork{
-				Common: executor.Common{
-					Name: contNetName,
-					ID:   moid,
-				},
-				Default: def,
-			},
-		})
-	} else {
-		conf.AddNetwork(&executor.NetworkEndpoint{
-			Network: executor.ContainerNetwork{
-				Common: executor.Common{
-					Name: contNetName,
-					ID:   moid,
-				},
-			},
-		})
+			Default:     def,
+			Nameservers: ns,
```

Comment:
* With DHCP on all networks, the user specified nameserver is currently ignored and only the DHCP provided nameserver ends up in `resolv.conf`. 

If this behavior is acceptable, I will add a warning that user input nameservers are ignored. If not, I'll see how `resolv.conf` is populated since we are currently passing the nameserver setting in the `NetworkEndpoint` but it's getting ignored in this case.

* With a static IP and user specified DNS servers and all networks on a single port group, we `resolv.conf` has only the user specified nameserver. 

* With a static IP on one network, DHCP on other networks, and a user specified nameserver, `resolv.conf` has both the user specified and DHCP provided nameservers.